actor/src/main/scala/org/apache/pekko/actor/dungeon/FaultHandling.scala

```diff
@@ -214,34 +214,34 @@ private[pekko] trait FaultHandling { this: ActorCell =>
   @InternalStableApi
   final def handleInvokeFailure(childrenNotToSuspend: immutable.Iterable[ActorRef], t: Throwable): Unit = {
     // prevent any further messages to be processed until the actor has been restarted
-    if (!isFailed)
```

Comment:
why is this check not needed?

Reply:
> why is this check not needed?

he move the check to match case

Reply:
the original code didn't use `setFailed()`, so we have no chance to update `actorRef`, and then we will get an assertion failed(the original issue)

This PR used `setFailed()` in exception handling, but we still need send a system message to ourself(to continue handling faults).

I think this check is the real problem that led to the original issue, and my PR removal check is to fix this problem.